### PR TITLE
Paid stats: add Stats reference on Personal checkout

### PR DESCRIPTION
**Pull request by Initsogar**
Resolves https://github.com/Automattic/red-team/issues/305

We want to include Stats features on the checkout page while purchasing a WordPress.com Personal plan.

To achieve this, we need to replace the existing feature texts from `FEATURE_STATS_PAID` with `FEATURE_STATS_COMMERCIAL` and add the new feature message for `FEATURE_STATS_PAID` on the checkout page.

To ensure a smooth transition, these changes are being made conditionally under the `stats/paid-wpcom-v3` feature flag to ensure everything works as expected.

CleanShot 2024-12-10 at 20 18 00@2x

> [!NOTE]
> Due to restrictions on how the features list is presented, I am merging both lines into a single line while we gather feedback from the design team. I believe it is safe to proceed with this PR and iterate if needed.
> | Expected | Proposal |
> |---|---|
> | CleanShot 2024-12-10 at 20 20 23@2x | CleanShot 2024-12-10 at 20 20 45@2x |

## Proposed Changes

* Add `FEATURE_STATS_COMMERCIAL` feature details using the existing `FEATURE_STATS_PAID`.
* Update `FEATURE_STATS_PAID` title/description with the new texts under the `stats/paid-wpcom-v3` feature flag.
* Replace `FEATURE_STATS_PAID` with conditional logic to use `FEATURE_STATS_COMMERCIAL` under the `stats/paid-wpcom-v3` feature flag in the Calypso plans list configuration.
  * This ensures that any references to `FEATURE_STATS_PAID` will continue working as usual and will switch to `FEATURE_STATS_COMMERCIAL` when the feature flag is enabled.

| Before | After |
|---|---|
| CleanShot 2024-12-10 at 19 47 04@2x | CleanShot 2024-12-10 at 19 47 32@2x |

## Why are these changes being made?

https://github.com/Automattic/red-team/issues/305

## Testing Instructions

* Spin up a Calypso Live Branch

### Validate Personal checkout has the new message under `stats/paid-wpcom-v3` flag

* Try upgrading to Personal plan. You can click on `Upgrade plan` on `Clicks` module.
* Ensure you see the `Stats beyond last 7 days` message in the features list (the feature flag is enabled by default in live branch environment).
  CleanShot 2024-12-10 at 20 01 56@2x
* Now disable the feature flag by appending `?flags=-stats/paid-wpcom-v3` to the checkout page.
* Ensure you don't see any Stats message in the features list.
  CleanShot 2024-12-10 at 19 59 55@2x
 
### Validate Premium checkout is not affected
* Try upgrading to Premium plan, you can click on `Upgrade plan` on the UTM or Devices module.
* In the checkout page, ensure you see  `In-depth site analytics dashboard.` in the features list, with and without the `stats/paid-wpcom-v3` flag.
  CleanShot 2024-12-10 at 19 54 17@2x

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by kangzj — APPROVED** [review]
Works as described 👍

File: packages/calypso-products/src/features-list.tsx [hunk 1]
```diff
@@ -1863,7 +1864,19 @@ const FEATURES_LIST: FeatureList = {
 	},
 	[ FEATURE_STATS_PAID ]: {
 		getSlug: () => FEATURE_STATS_PAID,
-		getTitle: () => i18n.translate( 'In-depth site analytics dashboard' ),
+		getTitle: () => {
+			return isEnabled( 'stats/paid-wpcom-v3' )
+				? i18n.translate( 'Stats beyond last 7 days. Detailed traffic stats and site insights.' )
+				: i18n.translate( 'In-depth site analytics dashboard.' );
```
Comment (lsl) [inline packages/calypso-products/src/features-list.tsx]:
```suggestion
				? i18n.translate( 'Stats beyond the last 7 days' )
				: i18n.translate( 'In-depth site analytics dashboard' );
```

File: packages/calypso-products/src/features-list.tsx [hunk 2]
```diff
@@ -1863,7 +1864,19 @@ const FEATURES_LIST: FeatureList = {
...
+		},
+		getDescription: () =>
+			i18n.translate(
+				'Deep-dive analytics and conversion data to help you make decisions to grow your site.'
+			),
+	},
+	[ FEATURE_STATS_COMMERCIAL ]: {
+		getSlug: () => FEATURE_STATS_COMMERCIAL,
+		getTitle: () => i18n.translate( 'In-depth site analytics dashboard.' ),
```
Comment (lsl) [inline packages/calypso-products/src/features-list.tsx]:
```suggestion
		getTitle: () => i18n.translate( 'In-depth site analytics dashboard' ),
```

I don't know why but these titles omit the full stops.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17057489

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @Initsogar for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17057489).

**Pull request merged**